### PR TITLE
k8s addons v2

### PR DESCRIPTION
**Pull request by calvix**
redesign of k8s-addons script

* moving to `kubectl`
* doing retries when `apply` fails
* moving files to array so we don't have to repeat same shit hundred times
ping @giantswarm/team-positive

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -658,70 +658,26 @@ write_files:
   permissions: 0544
   content: |
       #!/bin/bash
-      while ! curl --output /dev/null --silent --head --fail --cacert /etc/kubernetes/ssl/apiserver-ca.pem --cert /etc/kubernetes/ssl/apiserver-crt.pem --key /etc/kubernetes/ssl/apiserver-key.pem "https://{{.Cluster.Kubernetes.API.Domain}}:443"; do sleep 1 && echo 'Waiting for master'; done
-      curl -o /opt/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.6.1/bin/linux/amd64/kubectl 
+      # get kubectl
+      curl -o /opt/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.6.1/bin/linux/amd64/kubectl
```
Comment (rossf7) [inline templates.go]: Nitpick but any reason we don't use 1.6.4 here?

File: templates.go [hunk 2]
```diff
@@ -658,70 +658,26 @@ write_files:
...
+      for manifest in $MANIFESTS
+      do
+      		while
+         		/opt/bin/kubectl --kubeconfig=/etc/kubernetes/config/kubelet-kubeconfig.yml apply -f /srv/$manifest
+         		[ "$?" -ne "0" ]
+        	do
+                	echo "failed to apply /src/$manifest, retrying in 5 sec"
+                	sleep 5s
+        	done
+      done
```
Comment (kopiczko) [inline templates.go]: Spaces when nuts here

File: templates.go (hunk 1 above)
Comment (calvix) [inline templates.go]: good idea, can we somehow replace by some variable we have for this?

File: templates.go [hunk 3]
```diff
@@ -658,70 +658,26 @@ write_files:
...
-      curl -H "Content-Type: application/yaml" \
-        -XPOST -d"$(cat /srv/default-backend-dep.yml)" \
-        --cacert /etc/kubernetes/ssl/apiserver-ca.pem --cert /etc/kubernetes/ssl/apiserver-crt.pem --key /etc/kubernetes/ssl/apiserver-key.pem \
-        "https://{{.Cluster.Kubernetes.API.Domain}}:443/apis/extensions/v1beta1/namespaces/kube-system/deployments"
-      curl -H "Content-Type: application/yaml" \
-        -XPOST -d"$(cat /srv/default-backend-svc.yml)" \
-        --cacert /etc/kubernetes/ssl/apiserver-ca.pem --cert /etc/kubernetes/ssl/apiserver-crt.pem --key /etc/kubernetes/ssl/apiserver-key.pem \
-        "https://{{.Cluster.Kubernetes.API.Domain}}:443/api/v1/namespaces/kube-system/services"
+      # apply manifests
+      MANIFESTS="calico-configmap.yaml calico-node-sa.yaml calico-policy-controller-sa.yaml calico-ds.yaml calico-policy-controller.yaml kubedns-dep.yaml kubedns-svc.yaml default-backend-dep.yml default-backend-svc.yml ingress-controller-cm.yml ingress-controller-dep.yml ingress-controller-svc.yml"
```
Comment (rossf7) [inline templates.go]: This is really nice and cleans up the templates a lot <3

File: templates.go (hunk 1 above)
Comment (calvix) [inline templates.go]: ( i have no idea what is the name so feel free to replace or give me the name)

File: templates.go (hunk 2 above)
Comment (calvix) [inline templates.go]: yeah this is the `yaml` indentation vs `script indentation` hell :)

File: templates.go (hunk 1 above)
Comment (xh3b4sd) [inline templates.go]: This should actually be configured via the clustertpr. See https://github.com/giantswarm/kubernetesd/blob/a8528d72f2a2ff7b37c5d6674684a03c543830c1/kubernetes/configmap.yaml#L90. I realize now that this is not specific to KVM so the config should be moved in kubernetesd. Anyway, at the moment here the URL should be replaced by something like this. 

```
curl -o /opt/bin/kubectl {{ .Cluster.KVM.Kubectl.GoogleAPI.URL }}
```

Lets please create an issue for this and address it in the next steps.

Comment (calvix) [inline templates.go]: @xh3b4sd good idea, issues created https://github.com/giantswarm/k8scloudconfig/issues/136

Comment (rossf7) [inline templates.go]: @calvix I'd say `KUBECTL_VERSION` would work. For hyperkube etc the k8s version is specified in clustertpr. So I think the var can just be for kubectl.

**Pull request merged**